tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -141,83 +125,20 @@ def definition_func(inout_field: gtscript.Field[float]):
                 definition_func, "test_missing_nested_symbol", module, externals=externals
             )
 
-    def test_undecorated_delay(self, id_version):
-        A = 0
-
-        def undecorated_function():
-            return A
-
-        module = f"TestInlinedExternals_test_undecorated_delay_{id_version}"
-        externals = {"func": undecorated_function}
-
-        A = 1
-
-        # Direct function
-        def definition_func(inout_field: gtscript.Field[float]):
-            from gt4py.__gtscript__ import PARALLEL, computation, interval
-
-            with computation(PARALLEL), interval(...):
-                inout_field = undecorated_function()
-
-        stencil_id, def_ir = compile_definition(
-            definition_func, "test_undecorated_delay", module, externals=externals
-        )
-
-        stmt = def_ir.computations[0].body.stmts[0]
-        assert isinstance(stmt.value, gt_ir.ScalarLiteral) and stmt.value.value == 1
-
-        # As external
-        def definition_func(inout_field: gtscript.Field[float]):
-            from gt4py.__externals__ import func
-            from gt4py.__gtscript__ import PARALLEL, computation, interval
-
-            with computation(PARALLEL), interval(...):
-                inout_field = func()
-
-        stencil_id, def_ir = compile_definition(
-            definition_func, "test_undecorated_delay", module, externals=externals
-        )
-
-        stmt = def_ir.computations[0].body.stmts[0]
-        assert isinstance(stmt.value, gt_ir.ScalarLiteral) and stmt.value.value == 1
-
-    def test_function_import(self, id_version):
-        module = f"TestInlinedExternals_test_recursive_imports_{id_version}"
-
-        def some_function():
-            from __externals__ import const
-
-            return const
-
-        def definition_func(inout_field: gtscript.Field[float]):
-            from __externals__ import some_call
-
-            with computation(PARALLEL), interval(...):
-                inout_field = some_call()
-
-        stencil_id, def_ir = compile_definition(
-            definition_func,
-            "test_recursive_imports",
-            module,
-            externals={"some_call": some_function, "const": GLOBAL_CONSTANT},
-        )
-        assert set(def_ir.externals.keys()) == {
-            "some_call",
-            "const",
-        }
-
-    def test_recursive_imports(self, id_version):
+    def test_recursive_function_imports(self, id_version):
         module = f"TestInlinedExternals_test_recursive_imports_{id_version}"
 
+        @gtscript.function
         def func_nest2():
             from __externals__ import const
 
             return const
 
+        @gtscript.function
```

Comment:
I think a test is missing with functions calling nested functions for more than one level, to verify that externals are correctly propagated through different nesting levels.

Reply:
Good idea. To keep it simple, I just expanded the existing test.